bgc_data/nrs_derived_indices_data.sql

```diff
@@ -0,0 +1,283 @@
+-- Materialized view for the NRS Derived Indices product
+-- To be served as a WFS layer by Geoserver
+CREATE MATERIALIZED VIEW nrs_derived_indices_data AS
+WITH
+
+-- Zooplankton temp tables ------------------------------------------------------------------------
+zoopinfo_mod AS (
+    -- reclassify diet column in zoopinfo
+    SELECT taxon_name,
+           length_mm,
+           CASE WHEN zi.diet = 'Carnivore' THEN 'CC'
+                WHEN zi.diet = 'Omnivore' THEN 'CO'
+                WHEN zi.diet = 'Herbivore' THEN 'CO'
+               END AS diet
+    FROM zoopinfo zi
+),
+zoop_by_trip AS (
+    -- per-trip stats for all zooplankton
+    SELECT trip_code,
+           sum(zoop_abundance_m3) AS "ZoopAbundance_m3"
+    FROM bgc_zoop_raw
+    GROUP BY trip_code
+),
+copepods_by_trip AS (
+    -- per-trip stats for all copepods
+    SELECT zr.trip_code,
+           sum(zr.zoop_abundance_m3) AS "CopeAbundance_m3",
+           sum(zr.zoop_abundance_m3 * zi.length_mm) AS "CopeSumAbundanceLength",
+           sum(zr.zoop_abundance_m3) FILTER ( WHERE zi.diet = 'CC') AS "CC",
+           sum(zr.zoop_abundance_m3) FILTER ( WHERE zi.diet = 'CO') AS "CO"
+    FROM bgc_zoop_raw zr LEFT JOIN zoopinfo_mod zi USING (taxon_name)
+    WHERE zr.copepod = 'COPEPOD'
+    GROUP BY trip_code
+),
+copepod_species_by_trip_species AS (
+    -- zooplankton raw data filtered to include only correctly identified copepod species
+    -- taxon_name updated to identify each unique species
+    -- grouped by trip_code & taxon_name
+    SELECT trip_code,
+           genus || ' ' || substring(species, '^\w+') AS taxon_name,
+           sum(taxon_count) AS taxon_count
+    FROM bgc_zoop_raw
+    WHERE copepod = 'COPEPOD' AND
+          species != 'spp.' AND
+          species NOT LIKE '%cf.%' AND
+          species NOT LIKE '%/%' AND
+          species NOT LIKE '%grp%'
```

Comment:
```suggestion
          species NOT LIKE '%grp%' AND
          species NOT LIKE '%complex%'
```
We may need to add this filter, as we did for the CPR and NRS products

Reply:
Good spotting. @clairedavies I assume we do want to apply this filter here too?

Reply:
FYI this is only filtering the data used to calculate indices "NoCopepodSpecies_Sample", "ShannonCopepodDiversity" and "CopepodEvenness"